client/me/help/help-contact-form/index.jsx

```diff
@@ -532,6 +536,7 @@ export class HelpContactForm extends PureComponent {
 		}
 
 		if (
+			showAlternativeSiteOptionsField &&
```

Comment:
Maybe we can refactor this `if` and combine it with the next one to make it clear they are connected? Something like:

```
if ( helpSiteIsNotWpCom ) {
   if ( ! showAlternativeSiteOptionsField ) {
       // The site you’ve selected is a self-hosted WordPress site...
   } else if ( this.state.errorData !== 'jetpack_error' ) {
       // %(siteName)s may be a copy of WordPress with a different hosting service....
   }
}
```